gapis/replay/mapping_exporter.go

```diff
@@ -0,0 +1,150 @@
+// Copyright (C) 2019 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package replay
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"reflect"
+	"sort"
+
+	"github.com/google/gapid/core/data/binary"
+	"github.com/google/gapid/core/log"
+	"github.com/google/gapid/gapis/api"
+	"github.com/google/gapid/gapis/api/transform"
+	"github.com/google/gapid/gapis/memory"
+	"github.com/google/gapid/gapis/replay/builder"
+	"github.com/google/gapid/gapis/service"
+)
+
+type MappingExporter struct {
+	mappings []*service.VulkanHandleMappingItem
+	thread   uint64
+	path     string
+}
+
+func NewMappingExporter(ctx context.Context) *MappingExporter {
+	return &MappingExporter{
+		mappings: make([]*service.VulkanHandleMappingItem, 0, 0),
+		thread:   0,
+		path:     "",
+	}
+}
+
+func NewMappingExporterWithPrint(ctx context.Context, path string) *MappingExporter {
+	return &MappingExporter{
+		mappings: make([]*service.VulkanHandleMappingItem, 0, 0),
+		thread:   0,
+		path:     path,
+	}
+}
+
+func (m *MappingExporter) GetMappings() []*service.VulkanHandleMappingItem {
+	return m.mappings
+}
+
+func (m *MappingExporter) ExtractRemappings(ctx context.Context, s *api.GlobalState, b *builder.Builder, done func()) error {
+	var ret error
+
+	total := 0
+
+	for k, v := range b.Remappings {
+		typ := reflect.TypeOf(k)
+		var size uint64
+		if t, ok := k.(memory.SizedTy); ok {
+			size = t.TypeSize(s.MemoryLayout)
+		} else {
+			size = uint64(typ.Size())
+		}
+		if size != 1 && size != 2 && size != 4 && size != 8 {
+			// Ignore objects that are not handles
+			continue
+		}
+		// Count the number of actual Posts we expect
+		total += 1
+		func(k interface{}, typ reflect.Type, size uint64) {
```

Comment:
I know this is just moving code around, but we may have to be more intelligent here later, since this is a crazy number of posts. (posts will have a performance impact)

Reply:
I agree